File: packages/javascript/src/canary-branch.ts
/**
 * CANARY — do not merge.
 *
 * One-line branch. The colocated unit test exercises only the `x > 0`
 * side, so the LINE is 100% covered but a BRANCH is missed. If CI is
 * green, branch coverage is not enforced on changed code.
 */
export const canaryBranch = (x: number): number => (x > 0 ? 1 : 2);

File: packages/python/src/cachetta/canary_branch.py
"""CANARY — do not merge.

The unit test exercises only the ``x > 0`` path. Every LINE runs, so line
coverage is 100%, but the ``if`` has an untaken (False) arc, so branch
coverage is < 100%. If CI is green, branch coverage is not enforced on
changed code.
"""


def canary_branch(x: int) -> int:
    result = 2
    if x > 0:
        result = 1
    return result
